### DISCUSSION TITLE
Another question about slow prompt updates, hangs, gitstatusd in a big repo

### DISCUSSION THREAD

**Discussion by timkgh** [romkatv/powerlevel10k#2753] [General]
I understand p10k and gitstatusd are not getting further support, just trying to understand what my options are in their current shape ...

In a git repo with over 950K files (per `git ls-files`) the prompt takes over 10s to _update_ git status on a Macbook Pro M1. The prompt itself is fast, it returns quickly (~19ms per benchmark), except that sometimes it _appears_ to hang, e.g. an `ls` in a directory with very few files under the git repo can take seconds to return. I _think_ it's because `gitstatusd` pegs the CPUs, `top` shows 600-700% CPU use from `gitstatusd`, plus the 2nd process in `top` is some company security app I have no control over that probably monitors disk access and such that gets busy too when gitstatusd does its thing.

Setting `POWERLEVEL9K_VCS_MAX_INDEX_SIZE_DIRTY=100000` makes everything snappy again, but obviously I lose some info in the prompt and it's fine, I can live with it.

At this point I'm trying to understand whether there are better options. For example, I noticed that a plain `git status` uses 150% CPU and takes <1s to return (the repo is configured with fs monitor/watchman and other optimizations in .git/config to make git perform better in this big repo). I thought I'd be able to turn off gitstatusd via `POWERLEVEL9K_DISABLE_GITSTATUS=true` and have the prompt update faster if it used plain git directly. But then the prompt doesn't return at all for ~4s so that's not a solution. Is there no way to combine the async prompt with the plain `git status` which seems to be faster than gitstatusd in this case? I obviously don't understand all that's happening under the hood. Or have `gitstatusd` do less work, at the expense of only showing some of the info in the prompt (branch and what plain `git status` shows would be fine).

**Comment by romkatv** [comment]
> Is there no way to combine the async prompt with the plain `git status`

No, there isn't. Your best option is to use `POWERLEVEL9K_VCS_MAX_INDEX_SIZE_DIRTY`.
